### ISSUE TITLE
Certificate not saved as trusted when connecting to WPA Enterprise

### ISSUE THREAD

**Issue by DMaendlen** [GrapheneOS/os-issue-tracker#1982]
Hi,

tl;dr: when using WPA Enterprise with username / password GrapheneOS seems to be unable to save a certificate as "trusted" when I click on "trust on first use".

Prerequisites:

* WPA Enterprise with username / password via FreeRADIUS on OPNSense firewall
* FreeRADIUS cert signed by OPNSense CA

Action:
Click on "trust on first use"

Expectation:
GrapheneOS connects to WPA Enterprise network in the future, trusting the network.

Reality:
GrapheneOS does *not* trust the network, I have to manually change from the saved certificate to "trust on first use" again.

I have changed my phone's OS in the meantime but I'm a believer in "see something, say something" so I thought I'd drop the issue here. Maybe it'll be useful to the project.

Thank you all for your time and your hard work.

**Comment by thestinger** [comment]
It sounds like you aren't using a valid CA with a domain you control. That means the certificate isn't going to be trusted when reconnecting until you accept a prompt. The way this works is explained at https://source.android.com/docs/core/connect/wifi-tofu. Are you sure that this isn't simply working the standard way that it does on Android 13?

**Comment by r3g-5z** [comment]
https://github.com/GrapheneOS/os-issue-tracker/issues/1770#issuecomment-1339646004 is probably related.

**Comment by thestinger** [comment]
Seems to be a duplicate of that issue.

**Comment by DMaendlen** [comment]
> It sounds like you aren't using a valid CA with a domain you control. That means the certificate isn't going to be trusted when reconnecting until you accept a prompt. The way this works is explained at https://source.android.com/docs/core/connect/wifi-tofu. Are you sure that this isn't simply working the standard way that it does on Android 13?

Well, the CA cert is valid and I *did* install it but somehow it doesn't get saved as a valid CA cert. Either I'm doing something horrendously wrong (possible) or something's broken. I'll reevaluate the cert in the coming week and will get back to you no later than 2023-03-05.

**Comment by flawedworld** [comment]
@DMaendlen  Please provide us detailed information on how you generated/created the certificate(s).

**Comment by flawedworld** [comment]
Commands, OpenSSL versions etc
